### PR TITLE
chore(ts): Move typings to DefinitelyTyped, Close #68

### LINKED ISSUE #68: Move typings to DefinitelyTyped
Several of the Testing Library repositories are doing this in an effort to improve the type definitions (it's recommended by the TypeScript team to do this if the source isn't written with TypeScript).

Anyone wanna take that on? Can reference this PR: https://github.com/testing-library/react-testing-library/pull/437

### PR DESCRIPTION
**Pull request by simjes**
**What**:
Moved typings to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped).
See https://www.npmjs.com/package/@types/testing-library__cypress and DefinitelyTyped/DefinitelyTyped#37692

**Why**:
To close issue #68 

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,24 +78,13 @@ npm install --save-dev @testing-library/cypress
 
 ### With TypeScript
 
-Typings are defined under `@testing-library/cypress/typings`, and should be
```
Comment (kentcdodds) [inline README.md]: I love that people won't have to bother with this anymore.

**Comment by kentcdodds** [comment]
I think the build failure was a fluke. I hope so anyway.

**Comment by kentcdodds** [comment]
@all-contributors please add @simjes for code

**Comment by simjes** [comment]
Sweet, I hope so too 😄 Will send a PR to update testing-library/testing-library-docs as well

**Pull request merged**